rust/connlib/tunnel/src/device_channel/tun_windows.rs

```diff
@@ -127,16 +130,28 @@ impl Tun {
     // It's okay if this blocks until the route is added in the OS.
     pub fn add_route(&self, route: IpNetwork) -> Result<()> {
         tracing::debug!("add_route {route}");
-        let iface_idx = self.iface_idx;
-        // TODO: Pick a more elegant way to do this
-        Command::new("powershell")
-            .creation_flags(CREATE_NO_WINDOW)
-            .arg("-Command")
-            .arg(format!(
-                "New-NetRoute -InterfaceIndex {iface_idx} -DestinationPrefix \"{route}\""
-            ))
-            .stdout(Stdio::null())
-            .status()?;
+        let mut row = MIB_IPFORWARD_ROW2::default();
+        // Safety: Windows shouldn't store the reference anywhere, it's just setting defaults
+        unsafe { InitializeIpForwardEntry(&mut row) };
+
+        let prefix = &mut row.DestinationPrefix;
+        match route {
+            IpNetwork::V4(x) => {
+                prefix.PrefixLength = x.netmask();
+                prefix.Prefix.Ipv4 = SocketAddrV4::new(x.network_address(), 0).into();
+            }
+            IpNetwork::V6(x) => {
+                prefix.PrefixLength = x.netmask();
+                prefix.Prefix.Ipv6 = SocketAddrV6::new(x.network_address(), 0, 0, 0).into();
+                tracing::warn!("ipv6 not implemented right here yet");
```

Comment:
Does this means that this PR will break ipv6 routes right now?

I'd wait for the fix before merging

Reply:
actually I haven't been testing IPv6, it doesn't even work on main right now - #3119 

So it's not a regression per se but I'll see if it's easy to fix in this PR at the same time.

Reply:
#3120 has a proposed fix for IPv6 on Windows
They don't have any conflict so this could go ahead with broken IPv6, or it could wait on the fix.

Reply:
I tested with `curl --ipv6 -v https://ifconfig.net/ip` and it works. The IP address curl connects to, and the IP address ifconfig.net shows for me, both differ when the tunnel is up. So I removed this warning